credhub-format.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+set -eo pipefail
+
+# yq operates on yaml documents using a jq-like syntax.
+# For jq program syntax, see: https://stedolan.github.io/jq/manual/
+
+if [ -z $2 ]; then
+  echo "
+Usage: ./credhub-format.sh <source-file> <pipeline-name>
+Example: ./credhub-format.sh credentials.yml deploy-something > output.yml
+
+Read from a credential file and reformat the contents to a JSON format that CredHub
+can import, excluding any keys that do not appear in ci/pipeline.yml.
+
+Run this script from the root of the repository so it can find pipeline.yml. The
+output file is written to stdout. Additional output, such as the keys that were
+excluded, is written to stderr." >&2
+  exit 1
+fi
+
+# check silently if yq is installed.
+if ! which -s yq; then
+  echo "yq is required to run this script." >&2
+  exit 1
+fi
+
+if [ ! -f ci/pipeline.yml ]; then
```

Comment:
Not sure if it's better, but if we don't want to be dependent on the `ci/pipeline.yml` path convention, you could use `fly -t ci get-pipeline -p $pipelinename` to download the currently pipeline definition. Though I'm not sure if that has the credhub values already interpolated from variables or not

Reply:
I like that idea, but unfortunately the variables are already interpolated as you guessed.